### ISSUE TITLE
JEV-FB-02: Fallback - Calibration layer via temperature scaling

### ISSUE THREAD

**Issue by jmlusu** [jmlusu/light-speed-holdings#348]
## Task
Add calibration layer via temperature scaling on LLM logits for fallback routing.

## Approach
- Collect logits from LLM router on validation set
- Fit temperature scaling parameter (single scalar T)
- Apply: calibrated_prob = softmax(logits / T)
- Validate ECE on holdout set

## Implementation
```python
from scipy.optimize import minimize_scalar

def temperature_scaling(logits, labels):
    def ece_loss(T):
        probs = softmax(logits / T)
        return expected_calibration_error(probs, labels)
    
    result = minimize_scalar(ece_loss, bounds=(0.1, 10.0))
    return result.x

# At inference
calibrated_probs = softmax(logits / optimal_T)
```

## Acceptance Criteria
- [ ] Temperature parameter fitted on validation set
- [ ] ECE < 0.15 on holdout (target)
- [ ] Inference overhead < 5ms
- [ ] Persisted with model version

## Trigger
- JEV-010 decision = NO-GO
